NGS2_WITNESS_Cycle1_difi_prep_exp2.R

```diff
@@ -0,0 +1,218 @@
+connections.exp2.dcast <- dcast(connection.events,
+                                id+session+event ~ data.name,
+                                value.var="data.value")
+
+connections.exp2.final <- connections.exp2.dcast[0,]
+
+for (i in 1:length(connections.exp2.dcast$id))
+{
+  if
+  (connections.exp2.dcast$event[i]=='Connected'){
+    connections.exp2.final[i,] <- rbind(connections.exp2.dcast[i,])
+  }
+  else if
+  (connections.exp2.dcast$event[i]=='Disconnected'){
+    connections.exp2.final <-
+      subset(connections.exp2.final, !(
+        (playerId1 == connections.exp2.dcast[i, c("playerId1")] &
+           playerId2 == connections.exp2.dcast[i, c("playerId2")]) |
+          (playerId1 == connections.exp2.dcast[i, c("playerId2")] &
+             playerId2 == connections.exp2.dcast[i, c("playerId1")])
+      )
+      )
+  }
+  else {
+    NULL
+  }
+}
+
+connections.from.to.exp2 <- cbind(connections.exp2.final$playerId1,
+                                  connections.exp2.final$playerId2, 
+                                  connections.exp2.final$session)
+
+
+connections.from.to.exp2.final <- connections.from.to.exp2[complete.cases(connections.from.to.exp2), ]
+DIFI.1 <- merge (av.cooperation.exp2, score.exp2, by = "pid")
+
+DIFI.1.subset<-subset(DIFI.1, session == i)
+
+### Loop over each session to compute network metrics and graphs
+
+network.frame <- data.frame(matrix(ncol= 6))
+network.frame.names <-c("session", 
+                        "edge.density", 
+                        "reciprocity", 
+                        "transitivity", 
+                        "diameter", 
+                        "assortativity")
+colnames(network.frame) <- network.frame.names
+
+
+for(i in unique(connections.from.to.exp2.final[,3])) {
+  
+  connection.subset <- connections.from.to.exp2.final[connections.from.to.exp2.final[,3] == i,
+                                                      1:2]
+  DIFI.1.subset<-subset(DIFI.1, 
+                        session == i)
+  
+  net.exp2 <- graph_from_data_frame(d=connection.subset,
+                                    vertices=DIFI.1.subset,
+                                    directed=F)
+  
+  net.edge_density.exp2 <- edge_density(net.exp2, loops=F)
+  net.reciprocity.exp2 <- reciprocity(net.exp2)
+  net.transitivity.exp2 <- transitivity(net.exp2, type="global") # net is treated as an undirected network
+  net.diameter.exp2 <- diameter(net.exp2, directed=F, weights=NA)
+  net.assortativity_degree.exp2 <- assortativity_degree(net.exp2, directed=F)
+  network.row<-data.frame(cbind(i,
+                     net.edge_density.exp2, 
+                     net.reciprocity.exp2, 
+                     net.transitivity.exp2, 
+                     net.diameter.exp2, 
+                     net.assortativity_degree.exp2))
+  colnames(network.row) <- network.frame.names
+  network.frame<-rbind(network.frame, network.row)
+  
+  # Plot final network
+  
+  jpeg(file = paste('netgraph.', i, '.jpeg', sep = ''))

```

Comment:
Saving these as PDFs may give you better options later on if you were to use these in a deck, for example, as they'll scale better than a JPEG. However, this is a suggestion and no action necessary if you're fine with JPEGs.

Reply:
Absolutely, thanks, the jpegs were horrible.